### DISCUSSION TITLE
Using Visible Within a List View Breaks UI

### DISCUSSION THREAD

**Discussion by Farmbot26** [slint-ui/slint#2516] [General]
The following code make items in the ListView with visible == true overlap and bunch up at the top.

`
ListView {

     for item[i] in items-model : HorizontalBox {

          visible: item.should_show;
          text: item.text;

     }

}
`

**Comment by ogoffart** [comment]
I can reproduce the problem:


[Full testcase](https://slint-ui.com/releases/1.0.0/editor/?snippet=import+%7B+Button%2C+VerticalBox%2C+ListView%2C+HorizontalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++height%3A+200px%3B%0A++++width%3A+200px%3B%0A%0A++++property+%3C%5B%7Btext%3A+string%2C+should_show%3A+bool%7D%5D%3E+items-model%3A%0A++++++++%5B%0A++++++++++++%7Btext%3A+%22123%22%2C+should_show%3A+true+%7D%2C%0A++++++++++++%7Btext%3A+%22456%22%2C+should_show%3A+true+%7D%2C%0A++++++++++++%7Btext%3A+%22789%22%2C+should_show%3A+false+%7D%2C%0A++++++++++++%7Btext%3A+%22abc%22%2C+should_show%3A+true+%7D%2C%0A++++++++++++%7Btext%3A+%22def%22%2C+should_show%3A+true+%7D%2C%0A++++++++++++%7Btext%3A+%22ghi%22%2C+should_show%3A+true+%7D%2C%0A++++++++++++%7Btext%3A+%22jkl%22%2C+should_show%3A+true+%7D%2C%0A++++++++%5D%3B%0A++++ListView+%7B%0A++++++++for+item%5Bi%5D+in+items-model+%3A++HorizontalBox+%7B%0A++++++++++++visible%3A+item.should_show%3B%0A++++++++++++Text+%7B%0A++++++++++++++++text%3A+item.text%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0A%0A%0A)

**Reply by ogoffart** [reply]
I created https://github.com/slint-ui/slint/issues/2537 to track the bug

**Reply by ogoffart** [reply]
FYI, I added a workaround in https://github.com/slint-ui/slint/issues/2537#issuecomment-1505563235
